monitor/metrics/store.go

```diff
@@ -47,6 +47,15 @@ func (mtrs *Store) Add(m *api.Metric) {
 	window.Add(m)
 }
 
+// RemovePeer removes all metrics related to a peer from the Store.
+func (mtrs *Store) RemovePeer(pid peer.ID) {
+	mtrs.mux.Lock()
+	for _, mtrs := range mtrs.byName {
+		delete(mtrs, pid)
+	}
+	mtrs.mux.Unlock()
```

Comment:
`defer Unlock` above

Reply:
Overhead of defer is non-trivial, so since this doesn't have multiple returns, I preferred to leave as is.